lib/db/schema/patch-018-019.sql

```diff
@@ -0,0 +1,199 @@
+
+CREATE TABLE devices (
+  uid BINARY(16) NOT NULL,
+  id INT UNSIGNED NOT NULL,
+  sessionTokenId BINARY(32),
+  name VARCHAR(255) NOT NULL,
+  type VARCHAR(16) NOT NULL,
+  createdAt BIGINT UNSIGNED NOT NULL,
+  callbackURL VARCHAR(255),
+  INDEX devices_uid (uid),
+  PRIMARY KEY (uid, id)
+) ENGINE=InnoDB;
+
+
+CREATE PROCEDURE `createDevice_1` (
+    IN `inUid` BINARY(16),
+    IN `inSessionTokenId` BINARY(32),
+    IN `inName` VARCHAR(255),
+    IN `inType` VARCHAR(16),
+    IN `inCreatedAt` BIGINT UNSIGNED,
+    IN `inCallbackURL` VARCHAR(255)
+)
+BEGIN
+  INSERT INTO devices(
+    uid,
+    id,
+    sessionTokenId,
+    name,
+    type,
+    createdAt,
+    callbackURL
+  )
+  SELECT
+    inUid,
+    COALESCE(MAX(id), 0) + 1,
```

Comment:
Oh dear lord there's a whole section dedicated to “AUTO_INCREMENT Handling in InnoDB”...